x-pack/plugins/apm/server/routes/fleet/apm_java_agent_versions.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import fetch from 'node-fetch';
+
+const mavenJavaAgentUrl =
+  'https://repo1.maven.org/maven2/co/elastic/apm/elastic-apm-agent/maven-metadata.xml';
+
+const versionRegex = /<version>(\d+\.\d+\.\d+.*?)<\/version>/gm;
+const latestVersionRegex = /<latest>(\d+\.\d+\.\d+.*?)<\/latest>/;
```

Comment:
You don't need to parse out the latest version. The dropdown should just contain the literal string `latest` as the default option. The Java attacher has a special handling for the `latest` version and automatically downloads and attaches the latest agent.

Reply:
Does selecting a version such as 1.18.0.RC1 pass validation? See also https://github.com/elastic/kibana/issues/133374#issuecomment-1149701428

Reply:
Fixed in 3b938158dc96e8853254f6397e7719fa6b3f82ec